pkg/controller/migmigration/registry.go

```diff
@@ -53,3 +60,269 @@ func (t *Task) getAnnotations(client k8sclient.Client) (map[string]string, error
 	}
 	return annotations, nil
 }
+
+// Ensure the migration registries on both source and dest clusters have been created
+func (t *Task) ensureMigRegistries() (int, error) {
+	nEnsured := 0
+
+	plan := t.PlanResources.MigPlan
+	if plan == nil {
+		return nEnsured, nil
+	}
+	storage := t.PlanResources.MigStorage
+	if storage == nil {
+		return nEnsured, nil
+	}
+
+	clusters := t.getBothClusters()
+
+	for _, cluster := range clusters {
+		if !cluster.Status.IsReady() {
+			continue
+		}
+		client, err := cluster.GetClient(t.Client)
+		if err != nil {
+			return nEnsured, liberr.Wrap(err)
+		}
+
+		// Migration Registry Secret
+		secret, err := t.ensureRegistrySecret(client)
+		if err != nil {
+			return nEnsured, liberr.Wrap(err)
+		}
+
+		// Get cluster specific registry image
+		registryImage, err := cluster.GetRegistryImage(client)
+		if err != nil {
+			return nEnsured, liberr.Wrap(err)
+		}
+
+		// Migration Registry DeploymentConfig
+		err = t.ensureRegistryDeployment(client, secret, registryImage)
+		if err != nil {
+			return nEnsured, liberr.Wrap(err)
+		}
+
+		// Migration Registry Service
+		err = t.ensureRegistryService(client, secret)
+		if err != nil {
+			return nEnsured, liberr.Wrap(err)
+		}
+
+		nEnsured++
+	}
+
+	// Condition
+	ensured := nEnsured == 2 // Both clusters.
+	if ensured {
+		t.Owner.Status.SetCondition(migapi.Condition{
+			Type:     RegistriesCreated,
+			Status:   True,
+			Category: migapi.Required,
+			Message:  RegistriesCreatedMessage,
```

Comment:
I wonder if having incremental messages would be helpful.

1. ensure source registry, report successful
2. ensure destination registry, report successful.

That was if we are stuck on either source or destination it is clear where we are in the process. I know we have this pattern everywhere, but just a suggestion.

Reply:
I just added something like what you're suggesting in https://github.com/konveyor/mig-controller/pull/717/files#diff-7595a02f0293165f7556b448f21e75188e7f77f2b00a9fd05d999e39d777be1aR276-R281
 
This will let you know if we find a cluster with a missing registry pod and give the user a heads up. See the GIF i posted in the first comment for a demo.